test/boost/sstable_compaction_test.cc

```diff
@@ -2176,72 +2177,17 @@ class scrub_test_framework {
     }
 };
 
-SEASTAR_TEST_CASE(sstable_scrub_validate_mode_test) {
-    cql_test_config test_cfg;
+SEASTAR_THREAD_TEST_CASE(sstable_scrub_validate_mode_test) {
+    scrub_test_framework test;
 
-    auto& db_cfg = *test_cfg.db_config;
+    auto schema = test.schema();
 
-    // Disable cache to filter out its possible "corrections" to the corrupt sstable.
-    db_cfg.enable_cache(false);
-    db_cfg.enable_commitlog(false);
-
-    return do_with_cql_env([] (cql_test_env& cql_env) -> future<> {
-        return test_env::do_with_async([] (test_env& env) {
-            auto schema = schema_builder("ks", get_name())
-                    .with_column("pk", utf8_type, column_kind::partition_key)
-                    .with_column("ck", int32_type, column_kind::clustering_key)
-                    .with_column("s", int32_type, column_kind::static_column)
-                    .with_column("v", int32_type).build();
-            auto permit = env.make_reader_permit();
-
-            auto scrubbed_mt = make_lw_shared<replica::memtable>(schema);
-            auto sst = env.make_sstable(schema);
+    auto muts = tests::generate_random_mutations(test.random_schema(), 3).get();
```

Comment:
So above you used schema=test.schema(), but here you use test.random_schema(). Why? Shouldn't this be the same schema?

Reply:
Maybe the method names are a bit misleading. `random_schema()` returns the `tests::random_schema` instance, which is what `tests::generate_random_mutations()` wants. `schema()` just returns the `schema_ptr`.